File: code/namespace.ts
namespace Validation {
    const numberRegexp = /^[0-9]+$/;

    export interface StringValidator {
        isAcceptable(s: string): boolean;
    }

    export class ZipCodeValidator implements StringValidator {
        isAcceptable(s: string): boolean {
            return s.length === 5 && numberRegexp.test(s);
        }
    }
}

let validators: { [s: string]: Validation.StringValidator; } = {};
validators["ZipCode"] = new Validation.ZipCodeValidator();

File: code/namespace.kt
package validation

import kotlin.js.RegExp

val zipCodeRegex = RegExp("[0-9]+")

interface StringValidator {
    fun isAcceptable(s: String): Boolean;
}

class ZipCodeValidator : StringValidator {
    override fun isAcceptable(s: String): Boolean {
        return s.length == 5 && zipCodeRegex.test(s);
    }
}

